tests/util/test_parseVocabularySources.py

```diff
@@ -0,0 +1,63 @@
+import json
+import pytest
+
+from mock import mock_open
+
+from codalib import util
+
+
+def test_return_value(monkeypatch):
+    """
+    Test that parseVocabularySources returns a list of the correct length
+    where all elements are tuples.
+    """
+    read_data = json.dumps({
+        'terms': [
+            {'name': 'foo', 'label': 'Foo'},
+            {'name': 'bar', 'label': 'Bar'},
+            {'name': 'baz', 'label': 'Baz'},
+            {'name': 'qux', 'label': 'Qux'},
+        ]
+    })
+
+    monkeypatch.setattr('__builtin__.open', mock_open(read_data=read_data))
+    choices = util.parseVocabularySources('/foo/bar')
+
+    assert len(choices) == 4
+    # Verify that all element of the list are tuples.
+    assert all([type(choice) is tuple for choice in choices])
+
+
+def test_return_value_elements(monkeypatch):
+    """
+    Verify that the returned list elements contain the name and the label.
+    """
+    read_data = json.dumps({
+        'terms': [
+            {'name': 'foo', 'label': 'Foo'},
+        ]
+    })
+
+    monkeypatch.setattr('__builtin__.open', mock_open(read_data=read_data))
+    choices = util.parseVocabularySources('/foo/bar')
+    assert choices.pop() == ('foo', 'Foo')
+
+
+@pytest.mark.xfail
+def test_empty_file_does_not_raise_exception(monkeypatch):
+    """
+    Verify that that an exception will not be raised if the file is empty.
+    """
+    monkeypatch.setattr('__builtin__.open', mock_open())
+    util.parseVocabularySources('/foo/bar')
+
+
+@pytest.mark.xfail
+def test_empty_json_does_not_raise_exception(monkeypatch):
```

Comment:
hmmmmm..... well, upon closer inspection, I was going to suggest `test_file_empty_json_does_not_raise_exception`, but I think that is too verbose, and it seems your original is fine. I was thinking this was sort of a converse to the test above it, but they're kind of in the same vein of testing different empty things. So I think the original name is fine.